prow/cmd/webhook-server/helpers.go

```diff
@@ -144,25 +147,25 @@ func isCertValid(cert string) error {
 	return nil
 }
 
-func createSecret(client ClientInterface, ctx context.Context, expiry int, dns []string) (string, string, string, error) {
-	if err := client.CreateSecret(ctx, secretID); err != nil {
+func createSecret(client ClientInterface, ctx context.Context, o options) (string, string, string, error) {
```

Comment:
there is no unit test for this file, I think we should add them. The test could be done by mocking `ClientInterface`, for example:
```
type fakeClient struct {
  // map of: <secret_name>: {<version>: <value>}
  secrets map[string]map[string][]byte
}

func (f *fakeClient) CreateSecret(ctx context.Context, secretID string) error {
  if _, ok := f.secrets[secretID]; ok {
    return errors.New("secret already exist")
  }
  f.secrets[secretID] = make(map[string][]byte)
  return nil
}
func (f *fakeClient)	AddSecretVersion(ctx context.Context, secretName string, payload []byte) error {
}
func (f *fakeClient)	GetSecretValue(ctx context.Context, secretName string, versionName string) ([]byte, bool, error) {
}
```

Reply:
main_test.go has some of these tests. I think i could just add more tests here.

Reply:
oh i see now, these are actually different from the functions defined in the interface

Reply:
should I just test the functions defined in the ClientInterface or I should test these helpers seeing as the helpers basically call the functions in the ClientInterface

Reply:
There are some value testing the client that implements `ClientInterface`, but not that much as they are just very thin wrappers of the real client, for example in your code:
```
func (g *GCPClient) CreateSecret(ctx context.Context, secretID string) error {
	_, err := g.client.CreateSecret(ctx, secretID)
	if err != nil {
		return fmt.Errorf("could not create secret %v", err)
	}
	return nil
}
```

These helper functions have much more custom logics, which is where I think it makes sense to add unit tests here.